src/main/java/org/dataone/hashstore/filehashstore/FileHashStoreUtility.java

```diff
@@ -112,26 +121,135 @@ public static boolean isDirectoryEmpty(Path directory) throws IOException {
+
+        Path restorePath = parentPath.resolve(newFileName);
+        Files.move(pathToRename, restorePath, StandardCopyOption.ATOMIC_MOVE);
+    }
+
+    /**
+     * Delete all paths found in the given List<Path> object.
+     *
+     * @param deleteList Directory to check
+     */
+    public static void deleteListItems(Collection<Path> deleteList) {
+        ensureNotNull(deleteList, "deleteList", "deleteListItems");
+        if (!deleteList.isEmpty()) {
+            for (Path deleteItem : deleteList) {
+                if (Files.exists(deleteItem)) {
+                    try {
+                        Files.delete(deleteItem);
+                    } catch (Exception ge) {
+                        String warnMsg =
+                            "Attempted to delete metadata document: " + deleteItem + " but failed."
+                                + " Additional Details: " + ge.getMessage();
+                        logFHSU.warn(warnMsg);
+                    }
+
+                }
+            }
+        }
+    }
+
+    /**
+     * Checks whether a given string is empty or contains illegal characters, and throws an
+     * exception if so
      *
      * @param string   String to check
      * @param argument Value that is being checked
      * @param method   Calling method
-     * @throws IllegalArgumentException If the string is empty or null
+     * @throws IllegalArgumentException If the string is empty or contains illegal characters
      */
-    public static void checkForEmptyString(String string, String argument, String method)
+    public static void checkForEmptyAndValidString(String string, String argument, String method)
         throws IllegalArgumentException {
         ensureNotNull(string, "string", "checkForEmptyString");
         if (string.trim().isEmpty()) {
             String errMsg = "FileHashStoreUtility.checkForEmptyString - Calling Method: " + method
                 + "(): " + argument + " cannot be empty.";
             throw new IllegalArgumentException(errMsg);
         }
+        if (!isValidString(string)) {
+            String errMsg = "FileHashStoreUtility.checkForEmptyString - Calling Method: " + method
+                + "(): " + argument + " contains empty white spaces, tabs or newlines.";
+            throw new IllegalArgumentException(errMsg);
+        }
+    }
+
+    /**
+     * Iterates over a given string and checks each character to make sure that there are no
+     * whitespaces, tabs, new lines or other illegal characters.
+     * @param string String to check
+     * @return True if valid, False if illegal characters found.
+     */
+    public static boolean isValidString(String string) {
+        boolean valid = true;
+        for (int i = 0; i < string.length(); i++) {
+            char ch = string.charAt(i);
+            if (Character.isWhitespace(ch)) {
+                valid = false;
+                break;
+            }
+        }
+        return valid;
```

Comment:
I think something's wrong here. It would not nmake sense that `isBlank()` would fail for a single `"\n"`. I'm using isBlank() on the notification service, and it works fine -- [see the source code here](https://github.com/DataONEorg/notification-service/blob/65ffd08746803e5d69b8626db623c3298f3141c1/src/main/java/org/dataone/notifications/util/StringUtils.java#L17):

```java
    public static boolean isBlank(String str) {
        return str == null || str.isBlank();
    }
```

I even added [a testcase for that, in your honor 😁](https://github.com/DataONEorg/notification-service/blob/65ffd08746803e5d69b8626db623c3298f3141c1/src/test/java/org/dataone/notifications/util/StringUtilsTest.java#L21), which passes.

Writing our own version of something that already exists in the Java API not only takes time now, but also becomes a maintenance burden for our future selves (and makes it more complicated for others who have to read and understand the codebase)

Reply:
Sorry - I lacked accuracy in my statement. When the string is not empty, but contains a new line/illegal character - it doesn't catch this scenario. So I've replaced `.trim().isEmpty()` with `isBlank()`.

Reply:
What do you think? I agree with your statement though that we don't want to be writing our own version of something that already exists in the Java API.